src/s3access.cpp

```diff
@@ -110,37 +96,60 @@ QString S3Access::bucket() const
 
 S3ListResult *S3Access::list(QString const &prefix)
 {
-    char *date;
     QString signData;
     QString url;
+
     QNetworkReply *reply;
     S3ListResult *result;
 
-    date = s3_make_date();
-    signData = "GET\n\n\n" + QString(date) + "\n/" + m_bucket + "/";
     url = "http://" + m_baseUrl + "/" + m_bucket + "/?delimiter=/";
     if (!prefix.isEmpty()) {
         url += "&prefix=" + prefix;
     }
 
-    reply = performOp(GET, url, signData, date, nullptr, nullptr, nullptr);
+    reply = performOp(GET, url, nullptr, nullptr, nullptr);
     result = new S3ListResult(reply, this);
-    free(date);
 
     return result;
 }
 
-QNetworkReply *S3Access::performOp(Method method, QString const &url, QString const &sign_data, const char *date, QIODevice *in, const char *content_md5, const char *content_type)
+QNetworkReply *S3Access::performOp(Method method, QString const &url, QIODevice *in, const char *content_md5, const char *content_type, QString signDataKey)
 {
     QNetworkRequest request;
-    char *digest;
+    unsigned char digest[EVP_MAX_MD_SIZE +1 ]; // +1 for the \0 byte
```

Comment:
The +1 is no longer needed here (since you're not adding the null byte after all).